staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -342,6 +359,11 @@ var newETCD3Client = func(c storagebackend.TransportConfig) (*kubernetes.Client,
 		}
 		dialOptions = append(dialOptions, grpc.WithContextDialer(dialer))
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.EtcdGrpcHealthcheck) {
+		// ignore config set by etcd manual resolver inside etcd client
+		dialOptions = append(dialOptions, grpc.WithDisableServiceConfig())
```

Comment:
what exactly is this disabling? does this mean service configs returned by the server are now ignored? is that really what we want?

Reply:
It's disabling [this](https://github.com/etcd-io/etcd/blob/release-3.5/client/v3/internal/resolver/resolver.go#L43) config in etcd client. I think this is a trade off to not require etcd client update. See https://github.com/etcd-io/etcd/pull/16278#discussion_r1337889354
We can also update etcd client and remove exposure to service config changes. Probably good idea to have it in the all future clients anyway.